lambdas/transcode/index.py

```diff
@@ -75,16 +85,30 @@ def lambda_handler(request):
     except ValueError:
         return make_json_response(400, {'error': "Invalid 'duration'"})
 
+    format_params = []
+
+    if format in AUDIO_FORMATS:
+        format_params.extend([
+            '-b:a', '128k',  # 128KB audio bitrate
+            '-vn',  # Drop the video stream
+        ])
+    elif format in VIDEO_FORMATS:
+        format_params.extend([
+            "-vf", ','.join([
+                f"scale=w={width}:h={height}:force_original_aspect_ratio=decrease",
+                "crop='iw-mod(iw\\,2)':'ih-mod(ih\\,2)'",
+            ]),
+        ])
+    else:
+        assert False
+
     with tempfile.NamedTemporaryFile() as output_file:
         p = subprocess.run([
             FFMPEG,
             "-t", str(duration),
             "-i", url,
             "-f", format,
-            "-vf", ','.join([
-                f"scale=w={width}:h={height}:force_original_aspect_ratio=decrease",
-                "crop='iw-mod(iw\\,2)':'ih-mod(ih\\,2)'",
-            ]),
+            *format_params,
             "-timelimit", str(request.context.get_remaining_time_in_millis() // 1000 - 2),  # 2 seconds for padding
             "-fs", str(MAX_VIDEO_SIZE),
```

Comment:
would be useful to parametrize this, so we could use this **instead** of duration.

Use case: show list of videos each no more than 6Mb, no matter how long it is

Reply:
From man:

> Set the file size limit, expressed in bytes. No further chunk of bytes is written after the limit is exceeded. **The size of the output file is slightly more than the requested file size**.

so I don't think we could rely on it.

Reply:
It's more predictable than relying on duration+bitrate for setting expected filesize :)

Reply:
☝️ I agree with Max